src/symbolic/scoped_symbol.mli

```diff
@@ -0,0 +1,19 @@
+type scope_token =
+  | Open_scope of string
+  | Close_scope
+  | Symbol of Smtml.Symbol.t
+
```

Comment:
I'm thinking, we could have an opaque `Scoped_symbol.t` (which would be implemented as a `scope_token list`) and move a lot of the logic in this module, right?